docs/api/protocol.md

```diff
@@ -247,65 +247,70 @@ Unregisters the custom protocol of `scheme`.
 
 Returns `Boolean` - Whether `scheme` is already registered.
 
-### `protocol.interceptFileProtocol(scheme, handler)`
+### `protocol.interceptFileProtocol(scheme, handler[, filter])`
```

Comment:
Hm, I don't like the pattern of adding options as extra arguments. In general I think that "options" objects are a better pattern—they're self-documenting in code that uses them (since the options are named), and they're more extensible and flexible, especially if options are later removed.

There are a few possibilities here I think:
1. `protocol.interceptFileProtocol(scheme, handler[, options])` where `options` is `{ filter?: string[] }`. It's a little awkward to put the options after the handler though, as if specified inline the handler will be several lines long and the options object will be hard to spot in code that uses this.
2. `protocol.interceptFileProtocol(scheme[, options], handler)` as above, but put the optional argument before the handler function. A bit harder to handle in C++ but better ergonomically for the user.
3. `protocol.interceptFileProtocol(options, handler)` where `options` is `string | { scheme: string, filter?: string[] }`. Passing the first argument as a string is equivalent to passing `{ scheme: arg }`. This is similar to Node's [`http.request`](https://nodejs.org/api/http.html#http_http_request_options_callback).

I think either (2) or (3) is better than (1), and any of these would be an improvement.

Reply:
Is it ok to break API compatibility? I added the filters as the final argument to avoid breaking any existing apps that use intercept protocol.

Reply:
(2) wouldn't break API compatibility if correctly handled. The 2nd argument would be _optional_ — if not supplied, the behavior would be the same as before.

Reply:
I've implemented suggestion (2)

Reply:
Note that in the docs I've used `filter` as the argument name, to match what is used in web-request docs. Let me know if I should use `options` instead.

Reply:
The reason I suggested `options` is because we might add new options in future which aren't related to filtering.

Reply:
Ok, I will adjust the name and type